ShellyPy/gen1.py

```diff
@@ -46,6 +48,9 @@ def update(self):
 
         self.emeters = status.get("emeter", [])
 
+        # Request meter information
+        self.meter = self.meter(index)
+
```

Comment:
since the `/settings` endpoint doesn't return a list of meters, the best way to do it is guess.

```py
self.meters = []
meter_index = 0
while True:
    try:
        self.meter.append(self.meter(meter_index))
        meter_index += 1
    except:
        break
```

should do the job